src/lib/renderComponent.tsx

```diff
@@ -2,50 +2,81 @@ import * as cx from 'classnames'
 import * as React from 'react'
 import { FelaTheme } from 'react-fela'
 
+import callable from './callable'
+import felaRenderer from './felaRenderer'
 import getClasses from './getClasses'
 import getElementType from './getElementType'
 import getUnhandledProps from './getUnhandledProps'
-import callable from './callable'
+import toCompactArray from './toCompactArray'
+
+import {
+  ComponentStyleFunctionParam,
+  ComponentVariablesInput,
+  ComponentVariablesObject,
+  IComponentPartClasses,
+  IComponentPartStylesInput,
+  IProps,
+  IThemeInput,
+  IThemePrepared,
+} from '../../types/theme'
 
 export interface IRenderResultConfig<P> {
   ElementType: React.ReactType<P>
-  rest: { [key: string]: any }
-  classes: { [key: string]: string }
+  classes: IComponentPartClasses
+  rest: IProps
 }
 
 export type RenderComponentCallback<P> = (config: IRenderResultConfig<P>) => any
 
+type IRenderConfigProps = {
+  [key: string]: any
+  variables?: ComponentVariablesInput
+  styles?: IComponentPartStylesInput
+}
+
 export interface IRenderConfig {
   className?: string
   defaultProps?: { [key: string]: any }
   displayName?: string
   handledProps: string[]
-  props: { [key: string]: any }
-  rules?: { [key: string]: Function }
-  variables?: (siteVariables: object) => object
+  props: IRenderConfigProps
 }
 
 const renderComponent = <P extends {}>(
   config: IRenderConfig,
   render: RenderComponentCallback<P>,
 ): React.ReactNode => {
-  const { className, defaultProps, displayName, handledProps, props, rules, variables } = config
+  const { className, defaultProps, displayName, handledProps, props } = config
 
   return (
     <FelaTheme
-      render={theme => {
-        const { siteVariables = {}, componentVariables = {}, renderer } = theme
-
+      render={({
+        siteVariables = {},
+        componentVariables = {},
+        componentStyles = {},
+        rtl = false,
+        renderer = felaRenderer,
```

Comment:
See my previous comment about missing values for `rtl` and `renderer`. It would be a benefit to overall maintainability if the provider did all of the heavy lifting, and these consumers could be more confident about the interface they receive, since they don't have to work with all the same uncertainty.